rundoctests.py

```diff
@@ -9,11 +9,52 @@
 import os
 import sys
 import doctest
-import resource
+from doctest import DocTestParser, Example, SKIP
+from multiprocessing import Process
+if os.name != 'nt':
+    import resource
 
 flags = doctest.ELLIPSIS
+timeout = 600
+
+filenames = list(sys.argv[1:])
+if os.name == 'nt':
+    notinwindows = ['src/cysignals/pysignals.pyx',
+                    'src/cysignals/alarm.pyx',
+                    'src/cysignals/pselect.pyx']
+
+    for f in list(filenames):
+        if f in notinwindows:
+            filenames.remove(f)
+
+# Add an option to flag doctest what should not be run on windows.
+SKIP_WINDOWS = doctest.register_optionflag("SKIP_WINDOWS")
+SKIP_CYGWIN = doctest.register_optionflag("SKIP_CYGWIN")
+SKIP_POSIX = doctest.register_optionflag("SKIP_POSIX")
+
+flag_map = {
+    SKIP_WINDOWS: lambda: os.name == 'nt',
+    SKIP_CYGWIN: lambda: sys.platform == 'cygwin',
+    SKIP_POSIX: lambda: os.name == 'posix'
+}
+
+
+class CysignalsDocTestParser(DocTestParser):
```

Comment:
Note, I renamed this from its original name `SkipByOsDocTestParser` to something more general just in anticipation of possibly adding additional doctest flags particular for Cysignals.